analyzers/src/SonarAnalyzer.Common/Helpers/KnownType.Implementation.cs

```diff
@@ -24,100 +24,76 @@
 
 namespace SonarAnalyzer.Helpers
 {
-    internal abstract partial class KnownType
+    internal sealed partial class KnownType
     {
-        public string TypeName { get; private init; }
+        private readonly IList<string> namespaceParts;
+        private readonly string[] genericParameters;
 
-        internal abstract bool Matches(ITypeSymbol symbol);
+        public string TypeName { get; }
+        public string FullName { get; }
+        public bool IsArray { get; init; }
 
-        internal sealed class RegularKnownType : KnownType
+        internal KnownType(string fullName, params string[] genericParameters)
         {
-            private readonly IList<string> namespaceParts;
-            private readonly string[] genericParameters;
+            FullName = fullName;
+            this.genericParameters = genericParameters;
 
-            public bool IsArray { get; init; }
+            var parts = fullName.Split('.');
+            TypeName = parts[parts.Length - 1];
 
-            internal RegularKnownType(string fullTypeName, params string[] genericParameters)
-            {
-                this.genericParameters = genericParameters ?? Array.Empty<string>();
-
-                var parts = fullTypeName.Split('.');
-                TypeName = parts[parts.Length - 1];
-                namespaceParts = new ArraySegment<string>(parts, 0, parts.Length - 1);
-            }
+            namespaceParts = new ArraySegment<string>(parts, 0, parts.Length - 1);
+        }
 
-            internal override bool Matches(ITypeSymbol symbol) =>
-                IsMatch(symbol)
-                || IsMatch(symbol.OriginalDefinition);
+        internal bool Matches(ITypeSymbol symbol) =>
+            IsMatch(symbol)
+            || IsMatch(symbol.OriginalDefinition);
```

Comment:
Can be single line - forward https://github.com/SonarSource/sonar-dotnet/pull/5689#discussion_r887949927